backoff.go

```diff
@@ -0,0 +1,68 @@
+package deferred
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+const (
+	maxDelay = 1000 * time.Second
+	factor   = 1.6
+	maxInt64 = float64(math.MaxInt64 - 512)
+)
+
+// Strategy defines the algorithm for backoff
+type Strategy interface {
+	Backoff() time.Duration
+	Reset()
+}
+
+type retry struct {
+	delay time.Duration
+}
+
+func (r retry) Backoff() time.Duration {
+	return r.delay
+}
+
+func (r retry) Reset() {}
+
+type exponential struct {
+	initDelay time.Duration
+	retryNum  float64
+}
+
+func (ex *exponential) Backoff() time.Duration {
+	min := ex.initDelay
+	if min <= 0 {
+		min = 100 * time.Millisecond
+	}
+
+	if min >= maxDelay {
+		return maxDelay
+	}
+
+	//calculate this duration
+	minf := float64(min)
+	durf := minf * math.Pow(factor, ex.retryNum)
+	ex.retryNum++
+	durf = rand.Float64()*(durf-minf) + minf
+	//ensure float64 wont overflow int64
+	if durf > maxInt64 {
+		return maxDelay
+	}
+	dur := time.Duration(durf)
+	//keep within bounds
+	if dur < min {
+		return min
+	}
+	if dur > maxDelay {
+		return maxDelay
+	}
+	return dur
+}
+
+func (ex *exponential) Reset() {
```

Comment:
What would happen if noone would ever call this? Would it leak in some way? Or: do we really need the `reset()` method?

Reply:
Yeah, you are right, as we use one opts instance per one handler - we do not need to reset retry number. Will remove it